code/drasil-example/glassbr/lib/Drasil/GlassBR/ModuleDefs.hs

```diff
@@ -152,7 +152,7 @@ findCT = funcDef "find"
 
 extractColumnCT :: Func
 extractColumnCT = funcDef "extractColumn" "Extracts a column from a 2D matrix" 
-  [mat, j] (Vect Real) (Just "column of the given matrix at the given index")
+  [mat, j] (vect3D Real) (Just "column of the given matrix at the given index") -- TODO: is this correct?
```

Comment:
while the above `3D`s are likely right, I agree, this one is less clear. You'll have to see where `extractColumnCT` is used to see what the context says.

Reply:
Yeah, so this is to extract a given column in the matrix. It's used as part of this:

```haskell
interpMod :: Mod
interpMod = packmod "Interpolation" 
  "Provides functions for linear interpolation on three-dimensional data" []
  [linInterpCT, findCT, extractColumnCT, interpY, interpZ]
```

So yes, 3D.

Reply:
Right, so this is 'different' in the sense that this isn't a semantic operation but rather this is a programming-level operation where an array is used for packed storage and then its columns are extracted.

So in a separate language (`ModelExpr` ?), we need ways to talk about storage and concrete representations. It's a typical mistake to conflate these.

Reply:
@JacquesCarette should this separate language be within the scope of the project by @CSchank?  Ideally I would like us to include at least parts of this other language as needed for the current examples, but I don't have a feel for how difficult a task this is.

Reply:
It should be in-scope. It's a much simpler language.